src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(toBite) {
+    if (!(toBite instanceof Herbivore)) {
+      return;
+    }
+
+    if (toBite.hidden) {
+      return;
+    }
```

Comment:
[FUNCTION NAMING] - The `bite` method currently checks if the target is an instance of Herbivore and if it's not hidden before applying damage. However, the method should not return early if the target is another Carnivore. Instead, it should only check whether the target is a Herbivore and if it's not hidden. This will simplify the logic and adhere to the task requirements.